spec/models/principal_spec.rb

```diff
@@ -202,4 +202,325 @@
       expect(Firm.where(id: firm.id)).to be_empty
     end
   end
+
+  describe '#next_onboarding_action' do
+    context 'when principal has no parent firm' do
+      before :each do
+        principal.firm.destroy
```

Comment:
Yeah, we did wonder about this. Ultimately we included it for completeness. There are a few other permutations that are (currently) impossible. Happy to remove them if it's confusing though.